### PR TITLE
Replace some usages of LocationXY(Z)16

### PR DESCRIPTION
**Pull request by Gymnasiast**

### CODE REVIEW

File: src/openrct2/paint/VirtualFloor.cpp [hunk 1]
```diff
@@ -200,7 +198,7 @@ bool virtual_floor_tile_is_floor(int16_t x, int16_t y)
 }
 
 static void virtual_floor_get_tile_properties(
-    int16_t x, int16_t y, int16_t height, bool* outOccupied, bool* tileOwned, uint8_t* outOccupiedEdges, bool* outBelowGround,
+    CoordsXY loc, int16_t height, bool* outOccupied, bool* tileOwned, uint8_t* outOccupiedEdges, bool* outBelowGround,
```
Comment (tupaschoal) [inline src/openrct2/paint/VirtualFloor.cpp]: `const CoordsXY& loc`

File: src/openrct2/paint/VirtualFloor.cpp [hunk 2]
```diff
@@ -213,8 +211,8 @@ static void virtual_floor_get_tile_properties(
     // See if we are a selected tile
     if ((gMapSelectFlags & MAP_SELECT_FLAG_ENABLE))
     {
-        if (x >= gMapSelectPositionA.x && y >= gMapSelectPositionA.y && x <= gMapSelectPositionB.x
-            && y <= gMapSelectPositionB.y)
+        if (loc.x >= gMapSelectPositionA.x && loc.y >= gMapSelectPositionA.y && loc.x <= gMapSelectPositionB.x
+            && loc.y <= gMapSelectPositionB.y)
```
Comment (tupaschoal) [inline src/openrct2/paint/VirtualFloor.cpp]: Maybe we should introduce the `operator>=`

File: src/openrct2/paint/VirtualFloor.cpp [hunk 3]
```diff
@@ -177,9 +175,9 @@ bool virtual_floor_tile_is_floor(int16_t x, int16_t y)
 
     // Check if map selection (usually single tiles) are enabled
     //  and if the current tile is near or on them
-    if ((gMapSelectFlags & MAP_SELECT_FLAG_ENABLE) && x >= gMapSelectPositionA.x - _virtualFloorBaseSize
-        && y >= gMapSelectPositionA.y - _virtualFloorBaseSize && x <= gMapSelectPositionB.x + _virtualFloorBaseSize
-        && y <= gMapSelectPositionB.y + _virtualFloorBaseSize)
+    if ((gMapSelectFlags & MAP_SELECT_FLAG_ENABLE) && loc.x >= gMapSelectPositionA.x - _virtualFloorBaseSize
+        && loc.y >= gMapSelectPositionA.y - _virtualFloorBaseSize && loc.x <= gMapSelectPositionB.x + _virtualFloorBaseSize
+        && loc.y <= gMapSelectPositionB.y + _virtualFloorBaseSize)
```
Comment (tupaschoal) [inline src/openrct2/paint/VirtualFloor.cpp]: Would it look better like:
```c++
    const auto virtualFloorBaseCoords = CoordsXY{_virtualFloorBaseSize, _virtualFloorBaseSize};
    if ((gMapSelectFlags & MAP_SELECT_FLAG_ENABLE) && loc >= (gMapSelectPositionA - virtualFloorBaseCoords)
        && loc <= (gMapSelectPositionB + virtualFloorBaseCoords)
```
I'm unsure if putting it into `CoordsXY` are a bit of stretch

Comment (Gymnasiast) [reply src/openrct2/paint/VirtualFloor.cpp]: Another option would be to overload the CoordsXY + operator to also allow integers. But I don't think that should go into this PR, if I'm honest.

Comment (tupaschoal) [reply src/openrct2/paint/VirtualFloor.cpp]: I'm fine with it being done on another pr

Comment (duncanspumpkin) [reply src/openrct2/paint/VirtualFloor.cpp]: What would a + overload mean? Add integer to x coordinate, add to both x and y? It's a little ambiguous. I'm not 100% sure if the <= overloads make great sense either. What does it mean for a coordinate to be less than another.

Comment (ZehMatt) [reply src/openrct2/paint/VirtualFloor.cpp]: I'm with @duncanspumpkin, it adds some ambiguity, if theres some distance measurement involved it should be its own distance function outside of the class or make it a static to accept A and B as parameters.

**Pull request merged**